oap-server/analyzer/meter-analyzer/src/main/java/org/apache/skywalking/oap/meter/analyzer/prometheus/rule/Rules.java

```diff
@@ -40,36 +49,76 @@
+
+        final Path root = ResourceUtils.getPath(path);
+
+        Map<String, Boolean> formedEnabledRules = enabledRules
+                .stream()
+                .map(rule -> {
+                    rule = rule.trim();
+                    if (rule.startsWith("/")) {
+                        rule = rule.substring(1);
+                    }
+                    if (!rule.endsWith(".yaml") && !rule.endsWith(".yml")) {
+                        return rule + "{.yaml,.yml}";
                     }
-                    Rule rule = new Yaml().loadAs(r, Rule.class);
-                    rule.setName(fileName);
                     return rule;
-                } catch (IOException e) {
-                    LOG.debug("Reading file {} failed", f, e);
-                }
+                })
+                .collect(Collectors.toMap(rule -> rule, $ -> false));
+        List<Rule> rules;
+        try (Stream<Path> stream = Files.walk(root)) {
+            rules = stream
+                    .filter(it -> formedEnabledRules.keySet().stream()
+                                    .anyMatch(rule -> {
+                                        boolean matches = FileSystems.getDefault().getPathMatcher("glob:" + rule)
+                                                .matches(root.relativize(it));
+                                        if (matches) {
+                                            formedEnabledRules.put(rule, true);
+                                        }
+                                        return matches;
+                                    }))
+                    .map(pathPointer -> {
+                        Path relativize = root.relativize(pathPointer);
+                        String relativizePath = relativize.toString();
+                        relativizePath = relativizePath.substring(0, relativizePath.lastIndexOf("."));
+                        return getRulesFromFile(relativizePath, pathPointer);
+                    })
+                    .filter(Objects::nonNull)
+                    .collect(Collectors.toList()) ;
+        }
+
+        if (formedEnabledRules.containsValue(false)) {
+            List<String> rulesNotFound = formedEnabledRules.keySet().stream()
+                    .filter(rule -> !formedEnabledRules.get(rule))
+                    .collect(Collectors.toList());
+            throw new UnexpectedException("Some configuration files of enabled rules are not found, enabled rules: " + rulesNotFound);
+        }
+        return rules;
+    }
+
+    private static Rule getRulesFromFile(String ruleName, Path path) {
+        File file = path.toFile();
+        if (!file.isFile()) {
+            return null;
+        }
+        try (Reader r = new FileReader(file)) {
+            String fileName = file.getName();
+            if (fileName.startsWith(".")) {
                 return null;
-            })
-            .filter(Objects::nonNull)
-            .collect(Collectors.toList());
+            }
```

Comment:
Not sure why we need to skip hidden files, but if you insist, please use this 

```java
         if (!file.isFile() || file.isHidden()) {
             return null;
         }
```

Reply:
The hidden file would exist when we set the rule as `/folder/*`.(Such as .DS_store in MacOS) That was the first request.
But with we added `.yml` and `.yaml` to the match pattern, I think hidden file is not an issue.

Reply:
So should we skip hidden files for now?

Reply:
We could use Zhenxu's way to do so.